app/pages/[page].js

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import { applySession } from "next-session";
+import { useRouter } from "next/router";
+import { Forms, getHandler } from "../form-schema";
+import Button from "@button-inc/bcgov-theme/Button";
+
+export default function home({ formIndex, formData, validPage, prevPageUrl }) {
+  const Form = Forms[formIndex];
+  const router = useRouter();
+  const onFirstPage = prevPageUrl === -1;
+
+  const rerouteHandler = (nextPage, _isValid, lastPage) => {
+    router.push(lastPage ? "/end" : nextPage);
+  };
+
+  const handleBackClick = () => {
+    if (onFirstPage) return;
+    router.push(prevPageUrl);
+  };
+
+  return (
+    <>
+      {validPage && (
+        <Form formData={formData} rerouteHandler={rerouteHandler}>
+          {!onFirstPage && (
+            <Button type="button" variant="secondary" onClick={handleBackClick}>
+              Back
+            </Button>
+          )}
+          <Button variant="primary">Continue</Button>
+        </Form>
+      )}
+    </>
+  );
+}
+
+export async function getServerSideProps({ req, res }) {
```

Comment:
Left this mostly the same, just wanted to note getServerSideProps will only work in page components. Reason is it tells nextjs to actually make a request to the server and return stuff, so gets handled on page navigations only